app/src/main/java/com/mapbox/maps/testapp/examples/GlobeActivity.kt

```diff
@@ -0,0 +1,160 @@
+  private lateinit var infoTextView: TextView
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val binding = ActivityGlobeViewBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+
+    infoTextView = binding.infoText
+    mapboxMap = binding.mapView.getMapboxMap().apply {
+      setMapProjection(currentProjection)
+      loadStyle(
+        style(Style.TRAFFIC_DAY) {
+          +fillExtrusionLayer("3d-buildings", "composite") {
+            sourceLayer("building")
+            filter(
+              eq {
+                get("extrude")
+                literal("true")
+              }
+            )
+            minZoom(12.0)
+            fillExtrusionColor(Color.parseColor("#aaaaaa"))
+            fillExtrusionHeight(Expression.get("height"))
+            fillExtrusionBase(Expression.get("min_height"))
+            fillExtrusionOpacity(0.6)
+          }
+          +skyLayer("sky") {
+            skyType(SkyType.ATMOSPHERE)
+            skyAtmosphereSun(listOf(15.0, 89.5))
+          }
+          initLocationLayer(binding.mapView)
+        }
+      )
+    }
+
+    binding.switchProjectionBtn.setOnClickListener {
+      val newProjection = if (currentProjection === MapProjection.Globe) {
+        MapProjection.Mercator
+      } else {
+        MapProjection.Globe
+      }
+      mapboxMap.setMapProjection(newProjection)
+      currentProjection = newProjection
+      updateInfoText(mapboxMap.cameraState.zoom)
+    }
+
+    binding.flyToBtn.setOnClickListener {
+      with(binding.mapView.camera) {
+        val centerAnimator = createCenterAnimator(cameraAnimatorOptions(TARGET_POINT)) {
+          duration = ANIMATION_DURATION_MS / 2
+          interpolator = AccelerateDecelerateInterpolator()
+        }
+        val bearingAnimator = createBearingAnimator(cameraAnimatorOptions(TARGET_BEARING)) {
+          duration = ANIMATION_DURATION_MS
+          interpolator = AccelerateDecelerateInterpolator()
+        }
+        val zoomAnimator = createZoomAnimator(cameraAnimatorOptions(TARGET_ZOOM)) {
+          startDelay = ANIMATION_DURATION_MS / 2
+          duration = ANIMATION_DURATION_MS
+          interpolator = AccelerateDecelerateInterpolator()
+        }
+        val pitchAnimator = createPitchAnimator(cameraAnimatorOptions(TARGET_PITCH)) {
+          startDelay = ANIMATION_DURATION_MS / 4
+          duration = ANIMATION_DURATION_MS / 2
+          interpolator = LinearInterpolator()
+        }
+        playAnimatorsTogether(centerAnimator, bearingAnimator, zoomAnimator, pitchAnimator)
+      }
+    }
+
+    updateInfoText(mapboxMap.cameraState.zoom)
+    binding.mapView.camera.addCameraZoomChangeListener {
+      updateInfoText(it)
+    }
+  }
+
+  private fun initLocationLayer(mapView: MapView) {
+    mapView.location.apply {
+      enabled = true
+      locationPuck = LocationPuck3D(
+        modelUri = "asset://race_car_model.gltf",
```

Comment:
consider use an open source model instead.

Reply:
we will treat it all together at once in all examples as part of tailwork